tools/mason/MasonBuild.chpl

```diff
@@ -107,72 +107,64 @@ private proc checkChplVersion(lockFile : borrowed Toml) throws {
 proc buildProgram(release: bool, show: bool, force: bool, skipUpdate: bool,
                   ref cmdLineCompopts: list(string), tomlName="Mason.toml",
                   lockName="Mason.lock") throws {
+  const cwd = here.cwd();
+  const projectHome = getProjectHome(cwd, tomlName);
+  const toParse = open(projectHome + "/" + lockName, ioMode.r);
+  var lockFile = parseToml(toParse);
+  const projectName = lockFile["root"]!["name"]!.s;
 
-  try! {
+  // --fast
+  var binLoc = 'debug';
+  if release then
+    binLoc = 'release';
 
-    const cwd = here.cwd();
-    const projectHome = getProjectHome(cwd, tomlName);
-    const toParse = open(projectHome + "/" + lockName, ioMode.r);
-    var lockFile = parseToml(toParse);
-    const projectName = lockFile["root"]!["name"]!.s;
 
-    // --fast
-    var binLoc = 'debug';
-    if release then
-      binLoc = 'release';
+  // build on last modification
+  if projectModified(projectHome, projectName, binLoc) || force {
```

Comment:
You inherit the order here, but `force` should come first here. If `force` is used, we shouldn't need to evaluate `projectModified`.